### PR TITLE
twister: add option to ignore skipped tests in reports

### LINKED ISSUE #38179: twister: only report failures in merged junit output
**Is your enhancement proposal related to a problem? Please describe.**
Currently the merged junit report is 215 MB

**Describe the solution you'd like**
we should only report failures

**Describe alternatives you've considered**

**Additional context**
release meeting

### PR DESCRIPTION
**Pull request by nashif**
With --no-skipped, twister will skip over all tests marked as skipped
(filtered) in the junit report.

This is useful for CI where have 1000s of filtered tests that appear in
the report and in some cases cause tools parsing the output to fail or
provide incomplete results.

Fixes #38179

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

**Review by hakehuang — APPROVED** [review]

File: scripts/pylib/twister/twisterlib.py
Comment (ceolin) [inline scripts/pylib/twister/twisterlib.py:3656]: `or total != skips:` will not cause it showing some skipped tests ?

Comment (cfriedt) [inline scripts/pylib/twister/twisterlib.py:3656]: might be able to eliminate the double-negative if you rename the variable to `report_skipped`.

**Review by cfriedt — DISMISSED** [review]
Needs a rebase, but LGTM

**Review by PerMac — CHANGES_REQUESTED** [review]
Could you please add the same option to json and csv reports for consistency? Sorry for so late review, I missed this PR somehow.

**Review by PerMac — DISMISSED** [review]
I changed to approve as I don't wont to block this PR. However, the consistency in the reports is important, so please add such option also for json and csv

**Review by enjiamai — DISMISSED** [review]
I think we will need this feature, and this only enabled when --no-skipped used.

**Review by hakehuang — DISMISSED** [review]

**Comment by mbolivar-nordic** [comment]
@nashif please rebase

**Review by ceolin — DISMISSED** [review]

**Review by PerMac — APPROVED** [review]

**Review by enjiamai — APPROVED** [review]

**Pull request merged**
